pkg/controllers/user/networkpolicy/policy.go

```diff
@@ -0,0 +1,346 @@
+	logrus.Infof("pnplc Remove pnp=%+v", pnp)
+
+	projectIDNoColon, ok := pnp.Labels[projectIDNoColonLabel]
+	if !ok {
+		return pnp, fmt.Errorf("couldn't find projectIDNoLabel for pnp=%+v", *pnp)
+	}
+	logrus.Debugf("pnplc Updated projectIDNoColon: %v", projectIDNoColon)
+
+	set := labels.Set(map[string]string{projectIDNoColonLabel: projectIDNoColon})
+
+	err := pnplc.mgr.cluster.K8sClient.NetworkingV1().NetworkPolicies("").DeleteCollection(nil, v1.ListOptions{LabelSelector: set.String()})
+	if err != nil {
+		logrus.Errorf("error deleting network policies for projectIDNoColon=%v", projectIDNoColon)
+		return pnp, err
+	}
+	return pnp, nil
+}
+
+//func (plc *projectLifecycle) Create(p *v3.Project) (*v3.Project, error) {
+//	logrus.Infof("plc Create: %+v", *p)
+//	plc.createDefaultNetworkPolicy(p)
+//	return p, nil
+//}
+//
+//func (plc *projectLifecycle) Updated(p *v3.Project) (*v3.Project, error) {
+//	logrus.Infof("plc Updated: %+v", *p)
+//	return p, nil
+//}
+//
+//func (plc *projectLifecycle) Remove(p *v3.Project) (*v3.Project, error) {
+//	logrus.Infof("plc Remove: %+v", *p)
+//	// Need to delete the network  policy?
+//	return p, nil
+//}
+//
+//func (plc *projectLifecycle) createDefaultNetworkPolicy(p *v3.Project) error {
+//	return plc.mgr.createDefaultNetworkPolicy(p)
+//}
+
+func (ps *projectSyncer) Sync(key string, p *v3.Project) error {
+	if p == nil {
+		return nil
+	}
+	return ps.mgr.createDefaultNetworkPolicy(p)
+}
+
+func (mgr *mgr) createDefaultNetworkPolicy(p *v3.Project) error {
+	if p == nil {
+		return nil
+	}
+	existingPolicies, err := mgr.pnpLister.List(p.Name, labels.NewSelector())
+	if err != nil {
+		logrus.Errorf("error fetching existing project network policy: %v", err)
+		return nil
+	}
+	if len(existingPolicies) > 0 {
+		logrus.Debugf("default network policy for project=%v already exists: %+v", p.Name, existingPolicies)
+		return nil
+	}
+
+	pnpDesc := fmt.Sprintf("Default network policy for project %v", p.Name)
+	projectID := p.Namespace + ":" + p.Name
+	projectIDNoColon := strings.Replace(projectID, ":", "_", 1)
+	logrus.Infof("creating default network policy for project %v", p.Name)
+	_, err = mgr.cluster.Management.Management.ProjectNetworkPolicies("").Create(&v3.ProjectNetworkPolicy{
+		ObjectMeta: v1.ObjectMeta{
+			GenerateName: "projectnetworkpolicy-",
+			Namespace:    p.Name,
+			Annotations: map[string]string{
+				projectIDAnnotation: projectID,
+			},
+			Labels: labels.Set(map[string]string{projectIDNoColonLabel: projectIDNoColon}),
+		},
+		Spec: v3.ProjectNetworkPolicySpec{
+			Description: pnpDesc,
+			ProjectName: projectID,
+		},
+	})
+	if err == nil {
+		logrus.Infof("Successfully created default network policy for project: %v", p.Name)
+	}
+	return err
+}
+
+func (nslc *namespaceLifecycle) Create(ns *corev1.Namespace) (*corev1.Namespace, error) {
+	logrus.Infof("nslc Create: %+v", *ns)
```

Comment:
Can you move the namespace labelers thingy to a completely separate controller, not in the networkpolicy package.